### PR TITLE
rules: Allow for setting labels on LHS on scalars

### PR DESCRIPTION
**Pull request by brian-brazil**

### CODE REVIEW

**Comment by fabxc** [comment]
Some context, please.
Did this cause a panic for nil map before or so? Does this need a test then?

**Comment by brian-brazil** [comment]
It caused a panic for a nil map, added test.

File: rules/recording_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+				Metric:    model.Metric{"__name__": "labels", "foo": "bar"},
+			}},
+		},
+	}
+
+	for _, test := range suite {
+		rule := NewRecordingRule(test.name, test.expr, test.labels)
+		result, _ := rule.eval(now, engine)
+		if !reflect.DeepEqual(result, test.result) {
+			t.Fatalf("Error expected, %q got ", test.result, result)
```
Comment (juliusv) [inline rules/recording_test.go]: The punctuation and interpolation placeholders are off here.

Comment (brian-brazil) [inline rules/recording_test.go]: Fixed

Comment (juliusv) [inline rules/recording_test.go]: Don't see the change yet. Forgot force push?

Comment (brian-brazil) [inline rules/recording_test.go]: Forgot to commit :)

File: rules/recording_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+				Metric:    model.Metric{"__name__": "labels", "foo": "bar"},
+			}},
+		},
+	}
+
+	for _, test := range suite {
+		rule := NewRecordingRule(test.name, test.expr, test.labels)
+		result, _ := rule.eval(now, engine)
+		if !reflect.DeepEqual(result, test.result) {
+			t.Fatalf("Error: expected %q got %q", test.result, result)
```
Comment (juliusv) [inline rules/recording_test.go]: Comma after the first `%q`?

**Comment by juliusv** [comment]
:+1: otherwise

File: rules/recording_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+				Value:     1,
+				Timestamp: now,
+				Metric:    model.Metric{"__name__": "labels", "foo": "bar"},
+			}},
+		},
+	}
+
+	for _, test := range suite {
+		rule := NewRecordingRule(test.name, test.expr, test.labels)
+		result, _ := rule.eval(now, engine)
```
Comment (fabxc) [inline rules/recording_test.go]: No ignoring of errors.

Comment (brian-brazil) [inline rules/recording_test.go]: Fixed

File: rules/recording_test.go (hunk 2 above)
Comment (brian-brazil) [inline rules/recording_test.go]: Done

**Comment by fabxc** [comment]
:+1:

**Pull request merged**